compiler/rustc_expand/src/mbe/metavar_expr.rs

```diff
@@ -93,7 +93,16 @@ fn parse_count<'sess>(
     span: Span,
 ) -> PResult<'sess, MetaVarExpr> {
     let ident = parse_ident(iter, sess, span)?;
-    let depth = if try_eat_comma(iter) { Some(parse_depth(iter, sess, span)?) } else { None };
+    let depth = if try_eat_comma(iter) {
+        if iter.look_ahead(0).is_none() {
+            return Err(sess
+                .span_diagnostic
+                .struct_span_err(span, "invalid meta-variable `count` expression"));
```

Comment:
The wording here is entirely unhelpful, need to say something about what was expected.

Reply:
Thanks @petrochenkov